gwcs/wcs.py

```diff
@@ -328,11 +556,345 @@ def invert(self, *args, **kwargs):
         else:
             return result
 
-    def _invert(self, *args, **kwargs):
+    def _invert_vfp(self, *args, **kwargs):
         """
-        Implement iterative inverse here.
+        Implements vectorized iterative inverse using fixed-point iterations.
         """
-        raise NotImplementedError
+        tolerance = kwargs.get('tolerance', 1e-5)
+        maxiter = kwargs.get('maxiter', 50)
+        adaptive = kwargs.get('adaptive', True)
+        detect_divergence = kwargs.get('detect_divergence', True)
+        quiet = kwargs.get('quiet', True)
+        with_bounding_box = kwargs['with_bounding_box']
+        fill_value = kwargs['fill_value']
+
+        args_shape = np.shape(args)
+        nargs = args_shape[0]
+        arg_dim = len(args_shape) - 1
+
+        if nargs != self.world_n_dim:
+            raise ValueError("Number of input coordinates is different from "
+                             "the number of defined world coordinates in the "
+                             f"WCS ({self.world_n_dim:d})")
```

Comment:
I prefer the former. At least we will get a clear message instead of an unexpected crash. Is it possible to improve this check to distinguish between coordinate inputs and non-coordinate inputs?